scripts/ssl_tts/ssl_tts_vc.py

```diff
@@ -0,0 +1,254 @@
+        fmin=librosa.note_to_hz('C2'),
+        fmax=librosa.note_to_hz('C7'),
+        frame_length=1024,
+        hop_length=256,
+        sr=22050,
+        center=True,
+        fill_na=0.0,
+    )
+    pitch_contour = torch.tensor(f0, dtype=torch.float32)
+    _pitch_mean = pitch_contour.mean().item()
+    _pitch_std = pitch_contour.std().item()
+    if compute_mean_std:
+        pitch_mean = _pitch_mean
+        pitch_std = _pitch_std
+    if (pitch_mean is not None) and (pitch_std is not None):
+        pitch_contour = pitch_contour - pitch_mean
+        pitch_contour[pitch_contour == -pitch_mean] = 0.0
+        pitch_contour = pitch_contour / pitch_std
+
+    return pitch_contour
+
+
+def segment_wav(wav, segment_length=44100, hop_size=44100, min_segment_size=22050):
+    if len(wav) < segment_length:
+        pad = torch.zeros(segment_length - len(wav))
+        segment = torch.cat([wav, pad])
+        return [segment]
+    else:
+        si = 0
+        segments = []
+        while si < len(wav) - min_segment_size:
+            segment = wav[si : si + segment_length]
+            if len(segment) < segment_length:
+                pad = torch.zeros(segment_length - len(segment))
+                segment = torch.cat([segment, pad])
+
+            segments.append(segment)
+            si += hop_size
+        return segments
+
+
+def get_speaker_embedding(ssl_model, wav_featurizer, audio_paths, duration=None, device="cpu"):
+    all_segments = []
+    all_wavs = []
+    for audio_path in audio_paths:
+        wav = load_wav(audio_path, wav_featurizer)
+        segments = segment_wav(wav)
+        all_segments += segments
+        all_wavs.append(wav)
+        if duration is not None and len(all_segments) >= duration:
+            # each segment is 2 seconds with one second overlap.
+            # so 10 segments would mean 0 to 2, 1 to 3.. 9 to 11 (11 seconds.)
+            all_segments = all_segments[: int(duration)]
+            break
+
+    signal_batch = torch.stack(all_segments)
+    print("signal batch", signal_batch.shape)
+    # print("signal batch", signal_batch.shape)
+    signal_length_batch = torch.stack([torch.tensor(signal_batch.shape[1]) for _i in range(len(all_segments))])
+    # print("signal length", signal_length_batch.shape)
+    signal_batch = signal_batch.to(device)
+    signal_length_batch = signal_length_batch.to(device)
+    _, speaker_embeddings, _, _, _ = ssl_model.forward_for_export(
+        input_signal=signal_batch, input_signal_length=signal_length_batch, normalize_content=True
+    )
+
+    speaker_embedding = torch.mean(speaker_embeddings, dim=0)
+    l2_norm = torch.norm(speaker_embedding, p=2)
+    speaker_embedding = speaker_embedding / l2_norm
+
+    return speaker_embedding[None]
+
+
+def get_ssl_features_disentsngled(
+    ssl_model, wav_featurizer, audio_path, emb_type="embedding_and_probs", use_unique_tokens=False, device="cpu"
+):
+    wav = load_wav(audio_path, wav_featurizer)
```

Comment:
Please add some comments in this function to section it out for readability

Reply:
added some comments.

Reply:
Hm, I don't see any new comments--were the changes pushed?

Reply:
Pushed with more comments!

Reply:
Looks good, thank you!